EssentialApp/EssentialAppTests/CommentsUIIntegrationTests.swift

```diff
@@ -0,0 +1,267 @@
+		
+		sut.simulateUserTapOnErrorMessage()
+		XCTAssertEqual(sut.errorMessage, nil, "Expected error message to be hidden")
+	}
+	
+	func test_deinit_cancelRunningRequest() {
+		var sut: CommentsController?
+		var loader: LoaderSpy?
+		
+		autoreleasepool {
+			(sut, loader) = makeSUT()
+			sut?.loadViewIfNeeded()
+		}
+		
+		XCTAssertEqual(loader?.cancelCallCount, 0)
+		
+		sut = nil
+		
+		XCTAssertEqual(loader?.cancelCallCount, 1)
+	}
+	
+	// MARK: - Helpers
+	
+	private func makeSUT(file: StaticString = #filePath, line: UInt = #line) -> (sut: CommentsController, loader: LoaderSpy) {
+		let loader = LoaderSpy()
+		let sut = CommentsUIComposer.commentsComposedWith(commentsLoader: loader)
+		trackForMemoryLeaks(loader, file: file, line: line)
+		trackForMemoryLeaks(sut, file: file, line: line)
+		return (sut, loader)
+	}
+	
+	private func localized(_ key: String, file: StaticString = #file, line: UInt = #line) -> String {
+		let table = "ImageComments"
+		let bundle = Bundle(for: CommentsPresenter.self)
+		let value = bundle.localizedString(forKey: key, value: nil, table: table)
+		if value == key {
+			XCTFail("Missing localized string for key: \(key) in table: \(table)", file: file, line: line)
+		}
+		return value
+	}
+	
+	private func assert(_ sut: CommentsController, isRendering comments: [Comment], file: StaticString = #file, line: UInt = #line) {
+		guard sut.numberOfRenderedCommentViews == comments.count else {
+			return XCTFail("Expected \(comments.count) comments, got \(sut.numberOfRenderedCommentViews) instead", file: file, line: line)
+		}
+		comments.enumerated().forEach { index, comment in
+			assertThat(sut, hasViewConfiguredFor: comment, at: index, file: file, line: line)
+		}
+	}
+	
+	private func assertThat(_ sut: CommentsController, hasViewConfiguredFor comment: Comment, at index: Int, file: StaticString = #file, line: UInt = #line) {
+		let view = sut.commentView(at: index)
+		
+		guard let cell = view as? CommentCell else {
+			return XCTFail("Expected \(CommentCell.self) instance, got \(String(describing: view)) instead", file: file, line: line)
+		}
+		
+		let formatter = RelativeDateTimeFormatter()
+		let relativeDate = formatter.localizedString(for: comment.createdAt, relativeTo: Date())
+		
+		XCTAssertEqual(cell.author, comment.author, "author at index \(index)", file: file, line: line)
+		XCTAssertEqual(cell.date, relativeDate, "date at index \(index)", file: file, line: line)
+		XCTAssertEqual(cell.comment, comment.message, "message at index \(index)", file: file, line: line)
+	}
+	
+	private func makeComment(id: UUID = UUID(), message: String = "message", createdAt: Date = Date(), author: String = "author") -> Comment {
+		return Comment(id: id, message: message, createdAt: createdAt, author: author)
+	}
+	
+	class LoaderSpy: CommentsLoader {
+		private var completions = [(CommentsLoader.Result) -> Void]()
+		var loadCallCount: Int {
+			return completions.count
+		}
+		
+		var cancelCallCount = 0
```

Comment:
The setter should be private - this value should only be changed internally to prevent mistakes.

```suggestion
		private(set) var cancelCallCount = 0
```